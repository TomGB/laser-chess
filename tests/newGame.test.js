const chess = require('../src/')

describe('Laser Chess newGame setup', () => {
  const game = chess.newGame()

  it('creates a board with the pieces in the correct places', () => {
    expect(game.pieces).toMatchSnapshot()
  })

  it('has pieces of the correct colour in the correct place', () => {
    const pieceColours = game.pieces.map(row => row.map(piece => piece ? piece.colour.substring(0, 2) : null))

    expect(pieceColours).toEqual([
      ['re', null, null, null, 're', 're', 're', 're', null, null],
      [null, null, 're', null, null, null, null, null, null, null],
      [null, null, null, 'wh', null, null, null, null, null, null],
      ['re', null, 'wh', null, 're', 're', null, 're', null, 'wh'],
      ['re', null, 'wh', null, 'wh', 'wh', null, 're', null, 'wh'],
      [null, null, null, null, null, null, 're', null, null, null],
      [null, null, null, null, null, null, null, 'wh', null, null],
      [null, null, 'wh', 'wh', 'wh', 'wh', null, null, null, 'wh'],
    ])
  })

  it('creates a board with restrictions', () => {
    expect(game.boardRestrictions).toMatchSnapshot()
  })

  it('creates a game with the turn set to red', () => {
    expect(game.turn).toEqual('red')
  })

  it('creates a confirmation message', () => {
    expect(game.message).toEqual('game setup complete')
  })
})
