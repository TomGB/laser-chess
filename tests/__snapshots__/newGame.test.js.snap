// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Laser Chess newGame setup creates a board with restrictions 1`] = `
Array [
  Array [
    "none",
    "white",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "red",
    "white",
  ],
  Array [
    "red",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "white",
  ],
  Array [
    "red",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "white",
  ],
  Array [
    "red",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "white",
  ],
  Array [
    "red",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "white",
  ],
  Array [
    "red",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "white",
  ],
  Array [
    "red",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "white",
  ],
  Array [
    "red",
    "white",
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    undefined,
    "red",
    "none",
  ],
]
`;

exports[`Laser Chess newGame setup creates a board with the pieces in the correct places 1`] = `
Array [
  Array [
    Object {
      "colour": "red",
      "rotation": "D",
      "rotation2": undefined,
      "type": "laser",
      "x": 0,
      "y": 0,
    },
    null,
    null,
    null,
    Object {
      "colour": "red",
      "rotation": "D",
      "rotation2": undefined,
      "type": "block",
      "x": 4,
      "y": 0,
    },
    Object {
      "colour": "red",
      "rotation": undefined,
      "rotation2": undefined,
      "type": "king",
      "x": 5,
      "y": 0,
    },
    Object {
      "colour": "red",
      "rotation": "D",
      "rotation2": undefined,
      "type": "block",
      "x": 6,
      "y": 0,
    },
    Object {
      "colour": "red",
      "rotation": "RD",
      "rotation2": undefined,
      "type": "corner",
      "x": 7,
      "y": 0,
    },
    null,
    null,
  ],
  Array [
    null,
    null,
    Object {
      "colour": "red",
      "rotation": "DL",
      "rotation2": undefined,
      "type": "corner",
      "x": 2,
      "y": 1,
    },
    null,
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    Object {
      "colour": "white",
      "rotation": "LU",
      "rotation2": undefined,
      "type": "corner",
      "x": 3,
      "y": 2,
    },
    null,
    null,
    null,
    null,
    null,
    null,
  ],
  Array [
    Object {
      "colour": "red",
      "rotation": "UR",
      "rotation2": undefined,
      "type": "corner",
      "x": 0,
      "y": 3,
    },
    null,
    Object {
      "colour": "white",
      "rotation": "DL",
      "rotation2": undefined,
      "type": "corner",
      "x": 2,
      "y": 3,
    },
    null,
    Object {
      "colour": "red",
      "rotation": "UR",
      "rotation2": "DL",
      "type": "mirror",
      "x": 4,
      "y": 3,
    },
    Object {
      "colour": "red",
      "rotation": "RD",
      "rotation2": "LU",
      "type": "mirror",
      "x": 5,
      "y": 3,
    },
    null,
    Object {
      "colour": "red",
      "rotation": "RD",
      "rotation2": undefined,
      "type": "corner",
      "x": 7,
      "y": 3,
    },
    null,
    Object {
      "colour": "white",
      "rotation": "LU",
      "rotation2": undefined,
      "type": "corner",
      "x": 9,
      "y": 3,
    },
  ],
  Array [
    Object {
      "colour": "red",
      "rotation": "RD",
      "rotation2": undefined,
      "type": "corner",
      "x": 0,
      "y": 4,
    },
    null,
    Object {
      "colour": "white",
      "rotation": "LU",
      "rotation2": undefined,
      "type": "corner",
      "x": 2,
      "y": 4,
    },
    null,
    Object {
      "colour": "white",
      "rotation": "LU",
      "rotation2": "RD",
      "type": "mirror",
      "x": 4,
      "y": 4,
    },
    Object {
      "colour": "white",
      "rotation": "DL",
      "rotation2": "UR",
      "type": "mirror",
      "x": 5,
      "y": 4,
    },
    null,
    Object {
      "colour": "red",
      "rotation": "UR",
      "rotation2": undefined,
      "type": "corner",
      "x": 7,
      "y": 4,
    },
    null,
    Object {
      "colour": "white",
      "rotation": "DL",
      "rotation2": undefined,
      "type": "corner",
      "x": 9,
      "y": 4,
    },
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    Object {
      "colour": "red",
      "rotation": "RD",
      "rotation2": undefined,
      "type": "corner",
      "x": 6,
      "y": 5,
    },
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    Object {
      "colour": "white",
      "rotation": "UR",
      "rotation2": undefined,
      "type": "corner",
      "x": 7,
      "y": 6,
    },
    null,
    null,
  ],
  Array [
    null,
    null,
    Object {
      "colour": "white",
      "rotation": "LU",
      "rotation2": undefined,
      "type": "corner",
      "x": 2,
      "y": 7,
    },
    Object {
      "colour": "white",
      "rotation": "U",
      "rotation2": undefined,
      "type": "block",
      "x": 3,
      "y": 7,
    },
    Object {
      "colour": "white",
      "rotation": undefined,
      "rotation2": undefined,
      "type": "king",
      "x": 4,
      "y": 7,
    },
    Object {
      "colour": "white",
      "rotation": "U",
      "rotation2": undefined,
      "type": "block",
      "x": 5,
      "y": 7,
    },
    null,
    null,
    null,
    Object {
      "colour": "white",
      "rotation": "U",
      "rotation2": undefined,
      "type": "laser",
      "x": 9,
      "y": 7,
    },
  ],
]
`;
